manifests/repo.pp

```diff
@@ -14,21 +14,28 @@
       Yumrepo['powerdns'] -> Package <| title == $powerdns::authoritative_package_name |>
       Yumrepo['powerdns-recursor'] -> Package <| title == $powerdns::recursor_package_name |>
 
+      $repo_desc = $facts['os']['release']['major'] ? {
+        '8'     => 'PowerTools',
+        default => 'CRB',
+      }
+      $repo_name = $repo_desc.downcase()
       if ($facts['os']['name'] == 'Rocky') {
-        $mirrorlist = "https://mirrors.rockylinux.org/mirrorlist?arch=\$basearch&repo=PowerTools-\$releasever"
-        $gpgkey = 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial'
+        $code_repo_params = {
+          mirrorlist => "https://mirrors.rockylinux.org/mirrorlist?arch=\$basearch&repo=${repo_desc}-\$releasever",
```

Comment:
We might simplify the whole code by adding a parameter, which gets it's value from hiera.

Reply:
Good idea. I've updated the logic at least for the RHEL derivatives.

Reply:
After reading a bit about it, it seems to be more complicated.
Looking at https://wiki.rockylinux.org/rocky/repo/#base-repositories there's no `powertools` or `PowerTools` repo on Rocky 9, but instead it's `crb`.

`crb` is also available on `CentOS Stream 9`.

On Rocky 9 and CentOS Stream 9 we might use an exec to run `dnf config-manager --enable <repo>`

Reply:
Rocky 8 requires either the `dnf-plugins-core` or we just update the `enabled` line in `/etc/yum.repos.d/Rocky-PowerTools.repo` (might also be an option for the other versions :shrug: )

Reply:
I've found an even easier solution: the powertools repo isn't needed. EPEL is enough. Tested on CentOS Stream 9, Rocky 8 and 9

I'll create a PR to get rid of it.